### ISSUE TITLE
URIParser.unescape UTF-8 problem

### ISSUE THREAD

**Issue by samoconnor** [JuliaWeb/URIParser.jl#37]
`escape` seems to produce % escaped UTF-8, but `unescape` does not re-produce the original character.
```julia
julia> s = URIParser.escape("ü")
"%C3%BC"

julia> s = URIParser.unescape(s)
"Ã¼"
```

**Comment by samoconnor** [comment]
I think this alternate implementation of `unescape` is more correct:
```julia
function unescape(str::ASCIIString)
    r = UInt8[]
    l = length(str)
    i = 1
    while i <= l
        c = str[i]
        i += 1
        if c == '%'
            c = parse(UInt8, str[i:i+1], 16)
            i += 2
        end
        push!(r, c)
    end
   return bytestring(r)
end
```

```julia
julia> unescape(URIParser.escape("ü"))
"ü"
```

**Comment by malmaud** [comment]
Thanks, wanna make a PR?

**Comment by samoconnor** [comment]
Hi @malmaud.
Um, not just now I'm afraid. I have other pressing priorities.
However, I have a local diff using the `unescape` function above that fixes the SimpleDB problem I was having. At some point when I do a pass through my local diffs I'll create a PR, but that might not be fore a while...
Cheers,
Sam

**Comment by malmaud** [comment]
No rush.

**Comment by randyzwitch** [comment]
@samoconnor Are you still willing to make a PR? Would be good to fix this, get the package ready for 0.5, then tag a new version.

**Comment by samoconnor** [comment]
@randyzwitch see #38
